File: demo/folder-sync/Dockerfile.php
# cairn-folder-sync: P2P folder sync demo (PHP)
# Build from repo root: docker build -f demo/folder-sync/Dockerfile.php -t cairn-demo-folder-sync-php .

FROM php:8.3-cli-bookworm
WORKDIR /app

# Install composer
COPY --from=composer:2 /usr/bin/composer /usr/bin/composer

# Copy local cairn-p2p package
COPY packages/php/cairn-p2p /app/packages/php/cairn-p2p

# Copy demo files
COPY demo/folder-sync/php /app/demo/folder-sync/php

# Install dependencies
WORKDIR /app/demo/folder-sync/php
RUN composer install --no-interaction --no-dev 2>/dev/null || true

RUN useradd -r -s /bin/false cairn
USER cairn
ENTRYPOINT ["php", "cairn_sync.php"]

File: demo/folder-sync/Dockerfile.go
# cairn-folder-sync: P2P folder sync demo (Go)
# Build from repo root: docker build -f demo/folder-sync/Dockerfile.go -t cairn-demo-folder-sync-go .

# --- Builder stage ---
FROM golang:1.24-bookworm AS builder
WORKDIR /build
COPY . .
WORKDIR /build/demo/folder-sync/go
RUN go build -o cairn-folder-sync .

# --- Runtime stage ---
FROM debian:bookworm-slim
RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates && rm -rf /var/lib/apt/lists/*
RUN useradd -r -s /bin/false cairn
USER cairn
COPY --from=builder /build/demo/folder-sync/go/cairn-folder-sync /usr/local/bin/cairn-folder-sync
ENTRYPOINT ["cairn-folder-sync"]
